### PR TITLE
Added client-level REST helpers.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Adds client-level REST helpers.

This allows one to do `client.http.get(...)`, `.post`, etc., instead of `client.transport.perform_request("GET", ...)`.

### Issues Resolved

Closes https://github.com/opensearch-project/opensearch-py/issues/498.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@saimedhi @harshavamsi @Xtansia wdyt?

**Comment by Xtansia** [comment]
> @saimedhi @harshavamsi @Xtansia wdyt?

We're going to have this name clash with `get` in all the clients presumably, and it'd probably be nice to have this semi-standardised across them. An option is to make an "operation namespace" like `raw` or `http` same as we have `cat`/`cluster`/`indices` etc.

**Comment by dblock** [comment]
@Xtansia What's your most preferred option for `client.X.Y.Z`?

**Comment by Xtansia** [comment]
> @Xtansia What's your most preferred option for `client.X.Y.Z`?

I personally feel like `client.http.get()` is probably the most self-documenting, and least likely to ever conflict with the naming of a "real" operation

**Comment by dblock** [comment]
Followed suggestion from @Xtansia to make the code `client.http.get`, `client.http.post`, etc. This is ready to review!

File: guides/json.md [hunk 1]
```diff
@@ -6,61 +6,75 @@
 
 # Making Raw JSON REST Requests
 
-The OpenSearch client implements many high-level REST DSLs that invoke OpenSearch APIs. However you may find yourself in a situation that requires you to invoke an API that is not supported by the client. Use `client.transport.perform_request` to do so. See [samples/json](../samples/json) for a complete working sample.
+The OpenSearch client implements many high-level REST DSLs that invoke OpenSearch APIs. However you may find yourself in a situation that requires you to invoke an API that is not supported by the client. Use `client.http.get`, `head` , `put`, `post`, and `delete` in older versions to do so. See [samples/json](../samples/json) for a complete working sample.
```
Comment (VachaShah) [inline guides/json.md:9]: Use `client.transport.perform_request` in older versions (<= 2.3.x) and `client.http.get` and others in newer versions?

File: test_opensearchpy/test_async/test_client.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+#  Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (VachaShah) [inline test_opensearchpy/test_async/test_client.py:11]: Is this header needed?

**Review by VachaShah — COMMENTED** [review]
LGTM! Just a few minor comments.

File: test_opensearchpy/test_async/test_client.py (hunk 2 above)
Comment (dblock) [reply test_opensearchpy/test_async/test_client.py:11]: Yes, because the file was copied.

File: guides/json.md (hunk 1 above)
Comment (VachaShah) [reply guides/json.md:9]: I think "in older versions" would be removed from this line: "Use `client.http.get`, `head` , `put`, `post`, and `delete` to do so"?

Comment (dblock) [reply guides/json.md]: Good eyes, fixed.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**